plugins/GSdx/GSdx.rc

```diff
@@ -181,57 +181,63 @@ BEGIN
+    LTEXT           "Internal Resolution:",IDC_UPSCALE_MULTIPLIER_TEXT,22,144,79,8
+    COMBOBOX        IDC_UPSCALE_MULTIPLIER,105,142,127,98,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
+    LTEXT           "Custom Resolution:",IDC_CUSTOM_TEXT,22,160,79,8
+    EDITTEXT        IDC_RESX_EDIT,105,158,61,13,ES_AUTOHSCROLL | ES_NUMBER
+    CONTROL         "",IDC_RESX,"msctls_updown32",UDS_SETBUDDYINT | UDS_ALIGNRIGHT | UDS_AUTOBUDDY | UDS_ARROWKEYS | UDS_NOTHOUSANDS,145,158,11,14
+    EDITTEXT        IDC_RESY_EDIT,171,158,61,13,ES_AUTOHSCROLL | ES_NUMBER
+    CONTROL         "",IDC_RESY,"msctls_updown32",UDS_SETBUDDYINT | UDS_ALIGNRIGHT | UDS_AUTOBUDDY | UDS_ARROWKEYS | UDS_NOTHOUSANDS,221,158,11,14
+    LTEXT           "Texture Filtering:",IDC_FILTER_TEXT,22,176,79,8
+    COMBOBOX        IDC_FILTER,105,174,127,63,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
+    LTEXT           "Anisotropic Filtering:",IDC_AFCOMBO_TEXT,22,192,79,8
+    COMBOBOX        IDC_AFCOMBO,105,190,127,118,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
+    LTEXT           "CRC Hack Level:",IDC_CRC_LEVEL_TEXT,22,208,79,8
+    COMBOBOX        IDC_CRC_LEVEL,105,206,127,63,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
+    CONTROL         "Enable HW Hacks",IDC_HACKS_ENABLED,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,11,223,71,10
+    PUSHBUTTON      "Configure Hacks...",IDC_HACKSBUTTON,105,221,127,14
+    CONTROL         "Accurate Date",IDC_ACCURATE_DATE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,105,239,67,8
+    CONTROL         "Hardware Depth",IDC_TC_DEPTH,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,11,239,72,8
+    LTEXT           "Blending Unit Accuracy:",IDC_ACCURATE_BLEND_UNIT_TEXT,22,252,79,10
+    COMBOBOX        IDC_ACCURATE_BLEND_UNIT,105,251,127,63,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
+    CONTROL         "Logarithmic Z",IDC_LOGZ,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,11,239,90,8
+    CONTROL         "Alpha Correction",IDC_FBA,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,105,239,74,8
+    GROUPBOX        "Software Mode Settings",IDC_STATIC,6,269,231,25,BS_CENTER
+    CONTROL         "Edge Anti-aliasing (AA1)",IDC_AA1,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,11,279,91,10
+    LTEXT           "Extra rendering threads:",IDC_STATIC,116,280,83,8
+    EDITTEXT        IDC_SWTHREADS_EDIT,198,278,34,13,ES_AUTOHSCROLL | ES_NUMBER
+    CONTROL         "",IDC_SWTHREADS,"msctls_updown32",UDS_SETBUDDYINT | UDS_ALIGNRIGHT | UDS_AUTOBUDDY | UDS_ARROWKEYS | UDS_NOTHOUSANDS,221,277,11,14
+    GROUPBOX        "Custom Shader Settings",IDC_STATIC,6,295,231,94,BS_CENTER
+    CONTROL         "Enable Shade Boost",IDC_SHADEBOOST,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,11,307,90,10
+    PUSHBUTTON      "Configure Shade Boost...",IDC_SHADEBUTTON,105,305,127,14
+    CONTROL         "Enable External Shader",IDC_SHADER_FX,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,11,322,90,10
+    CONTROL         "Enable FXAA",IDC_FXAA,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,105,322,61,10
+    LTEXT           "External Shader",IDC_STATIC,22,337,52,8
```

Comment:
Well, I was thinking if disabling the external shader stuff depending on the shader checkbox state (and actually all the stuff in the hardware mode settings groupbox depending on whether a hardware renderer was selected) was necessary since you can press Home (and F9) to change the setting anyway. And then I forgot to do those labels ;)

I'll have it done soon.